strongbox-web-core/src/main/java/org/carlspring/strongbox/rest/UserController.java

```diff
@@ -0,0 +1,226 @@
+package org.carlspring.strongbox.rest;
+
+import org.carlspring.strongbox.users.domain.User;
+import org.carlspring.strongbox.users.service.UserService;
+
+import javax.inject.Inject;
+import java.io.IOException;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Optional;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.orientechnologies.orient.object.db.OObjectDatabaseTx;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cache.CacheManager;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.web.bind.annotation.*;
+
+@Controller
+@RequestMapping("/users")
+public class UserController
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(UserRestlet.class);
+
+    @Inject
+    UserService userService;
+
+    @Inject
+    OObjectDatabaseTx databaseTx;
+
+    @Inject
+    CacheManager cacheManager;
+
+    @Inject
+    ObjectMapper objectMapper;
+
+    // ----------------------------------------------------------------------------------------------------------------
```

Comment:
Could you please use Javadoc comments for this (everywhere you're commenting like this), so that it doesn't differ too much from the rest of the code base and so that you could get the Javadocs in the IDE...?